src/plugins/resolver/resolver.c

```diff
@@ -68,6 +80,16 @@ static void resolverInit (resolverHandle *p, const char *path)
 	p->tempfile = 0;
 
 	p->path = path;
+
+	{
+		pthread_mutexattr_t mutex_attr;
+
+		int result = pthread_mutexattr_init(&mutex_attr);
+		if(result == 0)
+			result = pthread_mutexattr_settype(&mutex_attr, PTHREAD_MUTEX_RECURSIVE);
+		if(result == 0)
+			result = pthread_mutex_init(getResolverMutex(), &mutex_attr);
```

Comment:
This seems not very robust. Failures of pthread_mutexattr_init are simply ignorend?

Additionally it seems like the mutex is reinitialized for every active resolver or did I miss something?